### PR TITLE
Change Polarion field which stores tmt id

### PR DESCRIPTION
**Pull request by KwisatzHaderach**
There is a requirement to use different field to store tmt id in Polarion.
This patch changes which field is used for any new imported/exported test case to newly created 'tmtuuid' field.
Any existing cases will still be found with existing ids in current field so no need to export again just for this change, just during the next export the id will be exported to the new field.

Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] include a release note

### CODE REVIEW

**Review by happz — APPROVED** [review]

File: tmt/convert.py [hunk 1]
```diff
@@ -897,13 +897,25 @@ def read_polarion_case(
...
+            if not dry_run:
+                polarion_case.tmtid = uuid
+                polarion_case.update()
+                # Check if it was really uploaded
+                polarion_case.tmtid = ''
+                polarion_case.reload()
+                if not polarion_case.tmtid:
+                    echo(style(
+                        f"Can't add ID because {polarion_case.project_id} project "
+                        "doesn't have tmtid field defined",
```
Comment (psss) [inline tmt/convert.py:917]:
```suggestion
                        "doesn't have the 'tmtid' field defined.",
```

File: tmt/export/polarion.py [hunk 2]
```diff
@@ -211,9 +211,18 @@ def export_to_polarion(test: tmt.base.Test) -> None:
...
+        polarion_case.tmtid = uuid
         polarion_case.update()  # upload the ID first so the case can be found in case of errors
-    echo(style(f"Append the ID {uuid}.", fg='green'))
+        # Check if it was really uploaded
+        polarion_case.tmtid = ''
+        polarion_case.reload()
+        if not polarion_case.tmtid:
+            echo(style(
+                f"Can't add ID because {polarion_case.project_id} project "
+                "doesn't have tmtid field defined",
```
Comment (psss) [inline tmt/export/polarion.py:222]:
```suggestion
                "doesn't have the 'tmtid' field defined.",
```

**Review by psss — APPROVED** [review]
Looks good, just two nitpicks.

**Comment by happz** [comment]
@KwisatzHaderach I'm afraid there's at least one failed polarion test in /plans/remote/polarion, can you take a look at it?

**Comment by KwisatzHaderach** [comment]
Fixed the tests (hopefully) here https://github.com/teemtee/tests/pull/8 which of these chickens/eggs we merge first?

**Comment by psss** [comment]
Rebased on the latest `main` and dropped the temporary hack to actually execute tests.

**Comment by happz** [comment]
Unrelated failures, merging.

**Pull request merged**